subsys/dfu/src/dfu_target_mcuboot.c

```diff
@@ -85,7 +144,16 @@ int dfu_target_mcuboot_init(size_t file_size)
 		return err;
 	}
 
-	offset = 0;
+	if (IS_ENABLED(CONFIG_SETTINGS_FCB)) {
+		err = settings_load();
+		if (err) {
+			LOG_ERR("Cannot register settings (err %d)", err);
+			return err;
+		}
+		flash_img.bytes_written = offset;
```

Comment:
Using an opaque type like this could be dangerous. 
What if we store the flash_img context object instead of just the offset, then read out the offset from the context using `flash_img_bytes_written` ?

Reply:
sure that could probably be better then we could get rid of the whole offset variable.